### PR TITLE
[Batch] Enable token credential for cloudshell

### PR DESCRIPTION
**Pull request by xingwu1**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by xingwu1** [comment]
@yugangw-msft @matthchr could you review my PR?

File: scripts/ci/test_extensions.sh [hunk 1]
```diff
@@ -21,7 +21,12 @@ exit_code=0
 for ext in $output; do
     echo
     echo "Verifying extension:" $ext
-    az extension add -n $ext
+    az extension add -n $ext --debug
```
Comment (yugangw-msft) [inline scripts/ci/test_extensions.sh]: is this change relevant?

File: src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/_client_factory.py [hunk 2]
```diff
@@ -75,8 +77,11 @@ def batch_data_service_factory(cli_ctx, kwargs):
     if not account_key:
         from azure.cli.core._profile import Profile
         profile = Profile(cli_ctx=cli_ctx)
+        # in order to use AAD auth in cloud shell mode, we will use mgmt AAD token
+        # instead of Batch AAD token to auth
         credentials, _, _ = profile.get_login_credentials(
-            resource=cli_ctx.cloud.endpoints.batch_resource_id)
+            resource=cli_ctx.cloud.endpoints.active_directory_resource_id
+            if in_cloud_console() else cli_ctx.cloud.endpoints.batch_resource_id)
```
[the remark this answers is not in the document]
Comment (yugangw-msft) [reply src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/_client_factory.py]: is using `active_directory_resource_id` is as good as `batch_resource_id`? if yes, then we just use the former one across.

[the remark this answers is not in the document]
Comment (xingwu1) [reply src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/_client_factory.py]: we will enable the AAD token based on appId. I don't think regular cli and cloud shell using same appId.

**Comment by troydai** [comment]
@xingwu1 is this PR still waiting for something?

**Comment by xingwu1** [comment]
@troydai I made mistake that the latest release server couldn't accept the the management endpoint AAD token. I have to wait for next release, which start at end of this week, and will be finished at next week.

**Comment by xingwu1** [comment]
@troydai  the service had been deployed, you can merge it.

**Comment by troydai** [comment]
@xingwu1 I would love to. Could you please bump the version of the command module.

**Pull request merged**